File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/SendReminderMailWorker.java
package io.miragon.blueprint.adapter.inbound.operaton;

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker;
import dev.bpmcrafters.processengine.worker.Variable;
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks;
import io.miragon.blueprint.application.port.inbound.SendSignatureReminderUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.springframework.stereotype.Component;

@Component
public class SendReminderMailWorker {

    private final SendSignatureReminderUseCase useCase;

    public SendReminderMailWorker(SendSignatureReminderUseCase useCase) {
        this.useCase = useCase;
    }

    @ProcessEngineWorker(topic = ServiceTasks.SEND_REMINDER_MAIL)
    public void sendReminderMail(@Variable String applicationId) {
        useCase.sendSignatureReminder(ApplicationId.of(applicationId));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/SendReminderMailWorker.kt
package io.miragon.blueprint.adapter.inbound.operaton

import dev.bpmcrafters.processengine.worker.ProcessEngineWorker
import dev.bpmcrafters.processengine.worker.Variable
import io.miragon.blueprint.adapter.process.BikeLeasingProcessProcessApi.ServiceTasks
import io.miragon.blueprint.application.port.inbound.SendSignatureReminderUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.springframework.stereotype.Component

@Component
class SendReminderMailWorker(
    private val useCase: SendSignatureReminderUseCase,
) {

    @ProcessEngineWorker(topic = ServiceTasks.SEND_REMINDER_MAIL)
    fun sendReminderMail(@Variable applicationId: String) {
        useCase.sendSignatureReminder(ApplicationId.of(applicationId))
    }
}
